frontend/device/kindle/powerd.lua

```diff
@@ -46,8 +46,17 @@ end
 function KindlePowerD:getCapacityHW()
     if self.lipc_handle ~= nil then
         return self.lipc_handle:get_int_property("com.lab126.powerd", "battLevel")
-    else
+    elseif self.batt_capacity_file then
         return self:read_int_file(self.batt_capacity_file)
+    else
+        local std_out = io.popen("gasgauge-info -c 2>/dev/null", "r")
+        if std_out then
+            local result = std_out:read("*all"):match("%d+")
+            std_out:close()
+            return result and tonumber(result) or 0
+        else
+            return 0
```

Comment:
Could potentially be easier to read if you just stick it at the end? (Not asking for a change, result is the same either way.)

Reply:
I'm not sure I get what you mean. Stick what at the end of what? ^^.

Reply:
I mean you can either write:

```lua
if bla then
  return lala
else
  return 0
end
```

or

```lua
if bla then
  lala
end
return 0
```

I find the latter slightly quicker to understand as a default fallback return value, but perhaps I just like that it's quicker to write. /shrugs

Reply:
So in this case right before the function's `end`.

Reply:
Oh ;).

Yeah, I'd have gone with the full verbose, explicit pyramid, but then I checked what `get_int_property` did, and I kind of liked it ;p.